larpandoracontent/LArMonitoring/HierarchyValidationAlgorithm.cc

```diff
@@ -376,11 +402,13 @@ void HierarchyValidationAlgorithm::MCValidation(const LArHierarchyHelper::MatchI
                 const LArHierarchyHelper::RecoHierarchy::NodeVector &nodeVector{matches.GetRecoMatches()};
                 const int nMatches{static_cast<int>(nodeVector.size())};
                 IntVector recoSliceIdVector, recoIdVector, nRecoHitsVector, nSharedHitsVector;
+		FloatVector recoTotalAdcVector;
                 FloatVector purityVector, completenessVector;
                 FloatVector purityAdcVector, completenessAdcVector;
                 FloatVector purityVectorU, purityVectorV, purityVectorW, completenessVectorU, completenessVectorV, completenessVectorW;
                 FloatVector purityAdcVectorU, purityAdcVectorV, purityAdcVectorW, completenessAdcVectorU, completenessAdcVectorV, completenessAdcVectorW;
-                const CartesianVector &trueVertex{pLeadingMC->GetVertex()};
+                const CartesianVector null(0.f,0.f,0.f);
+		const CartesianVector &trueVertex{isSignal == 1 ? pLeadingMC->GetVertex() : null};
```

Comment:
Given null is an absolute constant used in a couple of loops, I think I would move it's definition to the start of the function, perhaps after the definition of "interaction", and then it can be used in the two ternary operations without recreating the vector in every loop.